workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/androidx/StateRegistryAggregator.kt

```diff
@@ -82,9 +82,9 @@ internal class StateRegistryAggregator(
 
       // These properties are guaranteed to be non-null because this observer is only registered
       // while attached, and these properties are always non-null while attached.
-      val restoredState =
-        parentRegistryOwner!!.savedStateRegistry.consumeRestoredStateForKey(parentKey!!)
-      restoreFromBundle(restoredState)
+      parentRegistryOwner!!.savedStateRegistry.takeIf { it.isRestored }?.let {
+        restoreFromBundle(it.consumeRestoredStateForKey(parentKey!!))
```

Comment:
Probably should always call `restoreFromBundle` here, with `null` if necessary, so that its prune call always happens.